avocado/core/nrunner.py

```diff
@@ -576,20 +575,47 @@ def unittest_name(self):
             if module.startswith(os.path.sep):
                 module = module[1:]
         module = module.replace(os.path.sep, ".")
-        return '%s.%s' % (module, class_method)
+        klass, method = class_method.rsplit('.', maxsplit=1)
+        return module, klass, method
+
+    @property
+    def unittest_name(self):
+        """Convert a test reference (uri) to an unittest name reference."""
+        if not self.unittest:
+            return None
+        return "{}.{}.{}".format(*self.unittest)
+
+    @property
+    def module_path(self):
+        if not self.unittest:
+            return None
+        module, _, _ = self.unittest
+        path = module.replace('.', os.path.sep)
+        # This will fix adding support to hidden directories
+        path = path.replace('//', '/.')
+        return os.path.dirname(path)
+
+    @property
+    def module_class_method(self):
+        """Return a dotted name with module + class + method."""
+        if not self.unittest:
+            return None
+
+        return '.'.join(".".join(self.unittest).rsplit('.', maxsplit=3)[-3:])
```

Comment:
This is quite complex... would you care to explain what special situation it covers?  Like a sample input and output not covered by a simpler `".".join(self.unittest)[-3:]`?  Also, the mix of single and double quotes is a bit distracting :)

Reply:
You are right. In fact I will do a better refactoring here, there is no need for this complexity.